staging/src/k8s.io/apiserver/pkg/cel/environment/environment_test.go

```diff
@@ -251,6 +260,42 @@ func TestBaseEnvironment(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "recognizeKeywordAsFieldName disabled",
+			typeVersionCombinations: []envTypeAndVersion{
+				{version.MajorMinor(1, 30), NewExpressions},
+				// always enabled for StoredExpressions
+			},
+			invalidExpressions: []string{"gadget.namespace == 'buzz'"},
+			activation:         map[string]any{"gadget": map[string]any{"namespace": "buzz"}},
+			opts: []VersionedOptions{
+				{
+					IntroducedVersion: version.MajorMinor(1, 28),
+					DeclTypes:         []*apiservercel.DeclType{gadgetsType},
+					EnvOptions: []cel.EnvOption{
+						cel.Variable("gadget", cel.ObjectType("Gadget")),
+					},
+				},
+			},
+		},
+		{
+			name: "recognizeKeywordAsFieldName enabled",
+			typeVersionCombinations: []envTypeAndVersion{
+				{version.MajorMinor(1, 31), NewExpressions},
+				{version.MajorMinor(1, 30), StoredExpressions},
```

Comment:
```suggestion
				{version.MajorMinor(1, 32), NewExpressions},
				{version.MajorMinor(1, 31), StoredExpressions},
```

Is that we want? By merging this change, we introduce support for stored expressions into 1.31, but we don't support new expressions until 1.32 so that rollback to 1.31 is guaranteed safe

Reply:
The compatible env version for this release is set up to 1.30 which should be always one release behind. It is not the same version as the Kubernetes release. :)

Reply:
OK, yeah, I was thinking in terms of actual version, not compatibility version here.